### PR TITLE
refactor: css rewrite link

### PR DESCRIPTION
**Pull request by melaniebmn**
# Summary | Résumé

CSS rewrite for the link component to add new CSS features like layers, etc.

Part of fix for https://github.com/cds-snc/design-gc-conception/issues/627

### CODE REVIEW

File: packages/web/src/components/gcds-link/gcds-link.css [hunk 1]
```diff
@@ -1,49 +1,72 @@
...
     color: var(--gcds-link-light);
   }
+}
 
-  &.d-block {
-    display: block;
+@layer hover {
+  @media (hover: hover) {
+    :host .gcds-link:hover {
+      color: var(--gcds-link-hover);
```
Comment (ethanWallace) [inline packages/web/src/components/gcds-link/gcds-link.css:55]: When `gcds-link` has `variant="light"` the hover state still changes the text colour to dark blue making the text unreadable.

**Review by ethanWallace — CHANGES_REQUESTED** [review]
Just one small thing

File: packages/web/src/components/gcds-link/gcds-link.css (hunk 1 above)
Comment (melaniebmn) [reply packages/web/src/components/gcds-link/gcds-link.css:55]: Good catch, thank you!

**Review by ethanWallace — APPROVED** [review]
LGTM

**Pull request merged**